# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/html/nsGenericHTMLFrameElement.cpp
# Commit: 3b9e96b71449
# Full Hash: 3b9e96b714495177aa846bcee98c54e5f7a1c432
# Author: Paul Bone <pbone@mozilla.com>
# Date: 2019-09-10 15:48:08
# Regressor Bug: 1578628
# File Overlap Count: 1
# Description:
#   Bug 1578628 - Set the name on the BrowsingContext not the nsDocShell r=kmag
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D44947
# ==============================================================================

diff -r 43eace6b88a7 -r 3b9e96b71449 dom/html/nsGenericHTMLFrameElement.cpp
--- a/dom/html/nsGenericHTMLFrameElement.cpp	Tue Sep 10 07:05:07 2019 +0000
+++ b/dom/html/nsGenericHTMLFrameElement.cpp	Tue Sep 10 07:05:14 2019 +0000
@@ -353,15 +353,14 @@
         LoadSrc();
       }
     } else if (aName == nsGkAtoms::name) {
-      // Propagate "name" to the docshell to make browsing context names live,
-      // per HTML5.
-      nsIDocShell* docShell =
-          mFrameLoader ? mFrameLoader->GetExistingDocShell() : nullptr;
-      if (docShell) {
+      // Propagate "name" to the browsing context per HTML5.
+      RefPtr<BrowsingContext> bc =
+          mFrameLoader ? mFrameLoader->GetBrowsingContext() : nullptr;
+      if (bc) {
         if (aValue) {
-          docShell->SetName(aValue->String());
+          bc->SetName(aValue->String());
         } else {
-          docShell->SetName(EmptyString());
+          bc->SetName(EmptyString());
         }
       }
     }